src/service/viewer-impl.js

```diff
@@ -679,11 +691,14 @@ export class Viewer {
     log.fine(TAG_, 'message channel established with origin: ', origin);
     this.messageDeliverer_ = deliverer;
     // TODO(dvoytenko, #1764): Make `origin` required when viewers catch up.
-    this.messagingOrigin_ = origin;
     if (this.trustedViewerResolver_) {
       this.trustedViewerResolver_(
           origin ? this.isTrustedViewerOrigin_(origin) : false);
     }
+    if (this.messagingOriginResolver_) {
```

Comment:
So, let's revert `this.messagingOrigin_ = origin;` and instead add `this.viewerOriginResolver_(origin || '')`.